GetStream/stream-video-android
# Task/screen sharing support, UI work, landscape

### 🎯 Goal

Resolves #139 and introduces general improvemetns to the call participants UI and other parts of the call UI/flow.

Added the screensharing support based on the BE feature set. The changes are:
- Changed the way the CallParticipants UI works based on if there are screen sharing sessions or not.
    - Now shows the `RegularCallParticipantsContent` if there is no screenshare
    - Alternatively shows the `ScreenSharingCallParticipantsContent` if there is a screenshare session active
    - Handling of screen share state, UI changes, aspect ratio and so on

Track/State handling improvements
- Tracks are _tracked_ based on BE events and the AddStream function.
- Tracks are added when a screen share type stream arrives
- Tracks are removed when track unpublished events arrive
- ScreenShare sessions are tracked separately as there could be more than one
- Removed unused callbacks

Also made improvements to various UI components and state
- Changed the way the CallParticipants renders participants based on who is in the call
- Renders only the local participant (if we're the only participant).
- Renders a resizable set of participants (single, column of two, grid of four+...)
- Renders a floating participant item if there's more than 1 person in the call. Can be dragged around in the scope of the CallParticipants view.
- Various BG color changes and UI polish, like Participant label constraints

### 🛠 Implementation details

_Describe the implementation_

### 🎨 UI Changes

_Add relevant screenshots_

| Single Participant | Two participants | ScreenShare |
| --- | --- | --- |
| ![local_participant](https://user-images.githubusercontent.com/17215808/207828137-592a342b-8a27-4036-ac8b-5e6578e42c3f.png) | ![two_participants](https://user-images.githubusercontent.com/17215808/207828174-7b7e1143-68b0-44ca-abac-6c28c4c798d2.png)| ![screen_share](https://user-images.githubusercontent.com/17215808/207828223-4591cdb4-4445-4bd3-9e57-b7a334bcd231.png) |

<table>
<thead>
<tr>
<th>Screenshare demo</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<video src="https://user-images.githubusercontent.com/17215808/207828613-0148c793-4e21-4e15-82f3-e5b3462ea6f3.mp4" controls="controls" muted="muted" />
</tbody>
</table>

### 🧪 Testing

The only way to test screensharing is by connecting via Android(+/- more iOS/Android clients) and a Web client.
- Connect to the same call
- Initiatie Web client Screensharing
- Observe the state
- Stop screen sharing
- Observe the state

<details>

<summary>Provide the patch summary here</summary>

```
N/A for now
```

</details>


### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

Possible later checks:


### 🎉 GIF

![image](https://user-images.githubusercontent.com/17215808/207829871-f0af975e-cc6d-4a2e-939a-ad7c98336aad.png)
